### PR TITLE
Expose from_libusb() methods to import devices from Android

### PR DESCRIPTION
**Pull request by avantgardnerio**
Sorry for the 2nd PR, but I had to fight some libusb bugs to confirm this works, and it does!

With this PR, folks can use rust on Android as non-root users.

### CODE REVIEW

File: src/device_handle.rs [hunk 1]
```diff
@@ -139,13 +139,25 @@ impl<T: UsbContext> DeviceHandle<T> {
...
-            device::from_libusb(
+            device::Device::from_libusb(
                 self.context.clone(),
                 libusb_get_device(self.handle.as_ptr()),
             )
         }
     }
 
+    /// Unsafe function to convert an existing libusb_device_handle into a DeviceHandle<T>. Useful for Android.
+    pub unsafe fn from_libusb(
```
[the remark this answers is not in the document]
Comment (avantgardnerio) [reply src/device_handle.rs:150]: Updated. Hopefully correctly (I'm new at Rust). I'll also rebase the example app once this PR is approved and the API is stable.

File: src/context.rs [hunk 2]
```diff
@@ -81,7 +81,7 @@ pub trait UsbContext: Clone + Sized {
         if handle.is_null() {
             None
         } else {
-            Some(unsafe { device_handle::from_libusb(self.clone(), handle) })
+            Some(unsafe { DeviceHandle::from_libusb(self.clone(), std::ptr::NonNull::new_unchecked(handle)) })
```
Comment (a1ien) [inline src/context.rs:84]: small nitpick 
```suggestion
            Some(unsafe { DeviceHandle::from_libusb(self.clone(), NonNull::new(handle)?) })
```
And we can remove `if handle.is_null()`

**Review by a1ien — CHANGES_REQUESTED** [review]

File: src/context.rs (hunk 2 above)
Comment (avantgardnerio) [reply src/context.rs]: How about? ```        let ptr = std::ptr::NonNull::new(handle)?;
        Some(unsafe { DeviceHandle::from_libusb(self.clone(), ptr) })
```

**Comment by a1ien** [comment]
Thanks. Looks good. Can you squash commit with single message and force push

**Comment by avantgardnerio** [comment]
> Thanks. Looks good. Can you squash commit with single message and force push

:heavy_check_mark:

**Pull request merged**